src/Native/Runtime/MiscHelpers.cpp

```diff
@@ -778,9 +778,14 @@ EXTERN_C REDHAWK_API Int32 __cdecl RhpGetCurrentThreadStackTrace(void* pOutputBu
     return RhpCalculateStackTraceWorker(pOutputBuffer, outputBufferLength);
 }
 
-COOP_PINVOKE_HELPER(Boolean, RhpRegisterFrozenSegment, (void* pSegmentStart, UInt32 length))
+COOP_PINVOKE_HELPER(void*, RhpRegisterFrozenSegment, (void* pSegmentStart, size_t length))
 {
-    return RedhawkGCInterface::RegisterFrozenSection(pSegmentStart, length) != NULL;
+    return RedhawkGCInterface::RegisterFrozenSection(pSegmentStart, length);
```

Comment:
Nit: You can rename `RegisterFrozenSection` to `RegisterFrozenSegment` here while you are on it.

(Same for unregister.)

Reply:
Ok, done.